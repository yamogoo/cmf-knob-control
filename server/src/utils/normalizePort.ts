/**
 * Normalize a port.
 */

export default function normalizePort(val: string): number | undefined {
  const port = parseInt(val, 10)

  if (isNaN(port)) {
    // named pipe
    return Number(val)
  }

  if (port >= 0) {
    // port number
    return port
  }

  return undefined
}
