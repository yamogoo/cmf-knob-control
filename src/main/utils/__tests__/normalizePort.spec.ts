import { describe, expect, test } from 'vitest'

import normalizePort from '../normalizePort'

describe('normalizePort', () => {
  test.each(['0000', '9013', '1qdwq23'])('(%s) should be of type "number"', (port: string) => {
    const normalizedPort = normalizePort(port)
    const isNumber = typeof normalizedPort === 'number'

    expect(isNumber).toBe(true)
    expect(isNumber).toMatchSnapshot()
  })
})
