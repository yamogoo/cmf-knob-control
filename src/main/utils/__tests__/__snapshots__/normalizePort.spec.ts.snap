// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`normalizePort > (0000) should be of type "number" 1`] = `true`;

exports[`normalizePort > (1qdwq23) should be of type "number" 1`] = `true`;

exports[`normalizePort > (9013) should be of type "number" 1`] = `true`;
